scripts/list_hwm.py

```diff
@@ -0,0 +1,249 @@
+    sequence:
+      - type: map
+        mapping:
+          name:
+            required: false
+            type: str
+            desc: Name of the SoC
+          series:
+            required: false
+            type: str
+            desc: SoC series of the SoC.
+          family:
+            required: false
+            type: str
+            desc: SoC family of the SoC.
+          folder:
+            required: true
+            type: str
+            desc: Location of the SoC implementation relative to the socs.yml file.
+          vendor:
+            required: false
+            type: str
+            desc: SoC series of the SoC.
+                  This field is of informational use and can be used for filtering of SoCs.
+          arch:
+            required: false
+            type: str
+            desc: architecture of the SoC on a single-core SoC, for a multi-core SoC with
+                  mixed architectures, the value mixed can be used.
+                  This field is of informational use and can be used for filtering of SoCs.
+          comment:
+            required: false
+            type: str
+            desc: Free form comment with extra information regarding the SoC.
+'''
+
+ARCH_METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The archs.yml file is a simple list of key value pairs containing architectures
+# and their location which is used by the build system.
+type: map
+mapping:
+  archs:
+    required: true
+    type: seq
+    sequence:
+      - type: map
+        mapping:
+          name:
+            required: false
+            type: str
+            desc: Name of the arch
+          folder:
+            required: true
+            type: str
+            desc: Location of the arch implementation relative to the archs.yml file.
+          comment:
+            required: false
+            type: str
+            desc: Free form comment with extra information regarding the arch.
+'''
+
+SOCS_YML_PATH = PurePath('soc/socs.yml')
+ARCHS_YML_PATH = PurePath('arch/archs.yml')
+
+soc_schema = yaml.safe_load(SOC_METADATA_SCHEMA)
+arch_schema = yaml.safe_load(ARCH_METADATA_SCHEMA)
+
+
+def find_v2_archs(args):
+    ret = {'archs': []}
+    for root in args.arch_roots:
+        archs_yml = root / ARCHS_YML_PATH
+
+        if Path(archs_yml).is_file():
+            with Path(archs_yml).open('r') as f:
+                archs = yaml.safe_load(f.read())
+
+            try:
+                pykwalify.core.Core(source_data=archs, schema_data=arch_schema).validate()
+            except pykwalify.errors.SchemaError as e:
+                sys.exit('ERROR: Malformed "build" section in file: {}\n{}'
+                         .format(archs_yml.as_posix(), e))
+
+            if args.arch is not None and args.series is not None:
+                archs = {'archs': list(filter(
+                    lambda arch: arch.get('name') == args.arch, archs['archs']))}
```

Comment:
I don't know python but I'm going to assume this is an exact match, so "arm" will get you only `arm` boards (not `arm64`), and "aRm" will get you nothing? Or does is support wildcards e.g. "arm*" getting both `arm` and `arm64`?

Reply:
it's an exact match.